.github/workflows/build.yml

```diff
@@ -67,6 +67,15 @@ jobs:
         run: |
           bin/rake db:create db:schema:load
 
+      - name: Restore cached public assets
+        id: cache-public-restore
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            public/assets
+            public/packs-test
```

Comment:
The asset filenames contain fingerprints. In order to know if the most up-to-date assets are present, Rails is computing those fingerprints from the sources which involves most of the compilation. Repetitions are fast because many steps are cached in `tmp`. I don't know the exact steps and paths but I found that I needed to delete the whole tmp directory to trigger asset compilation from the beginning. So that would need to be cached, too. We would need a mechanism to purge that cache from time to time though. Otherwise we'll just pile up tmp files and multiple fingerprinted versions of the compiled assets.

Reply:
Thanks for looking into it.

> We would need a mechanism to purge that cache from time to time though. 

Oh yes, I was thinking about that initially but then ignored it ;)

I have a thought on that.. but I don't want us to build a custom solution. As much as I'm tempted to micro-optimise, we should avoid doing anything unless there's a pre-built solution

Reply:
I couldn't help but look a bit more though. Looks like you were right. I found some examples in the wild which might be enough:

1. Sprockets: https://gist.github.com/pftg/181813564423ce969e769fc6dfb72f70
   ```
            - public/assets
            - tmp/cache/assets/sprockets
   ```

2. Webpacker:  cache `tmp/cache/webpacker `and  `public/packs-test` separately: 
https://stackoverflow.com/questions/50883620/webpacker-ci-caching-required-folders


But it's time to cook pizza.